libs/island-ui/core/src/lib/DatePicker/DatePicker.tsx

```diff
@@ -92,8 +93,9 @@ export const DatePicker: React.FC<DatePickerProps> = ({
             [styles.popperWithoutLabel]: size === 'md' && !label,
           })}
           id={id}
+          name={name}
           disabled={disabled}
-          selected={selected ?? startDate}
+          selected={startDate ?? selected}
```

Comment:
This is so you can edit dates without having to control it from the parent.

Reply:
~Maybe it should have a corresponding "selected" vs "initialSelected" pattern instead of the currently confusing "startDate" state and double rendering with useEffect.

Reply:
~Maybe it should have a corresponding "selected" vs "initialSelected" pattern instead of the currently confusing "startDate" state and double rendering with useEffect.